### PR TITLE
[ISSUE-48][FEATURE][FOLLOW UP] Add RemoteShuffleService CRD

### PR DESCRIPTION
**Pull request by wangao1236**
### What changes were proposed in this pull request?
for issue #48 
I add a crd object called RemoteShuffleService this time, and I will submit the controller of the crd in next pr.

### CODE REVIEW

**Comment by jerqi** [comment]
@zuston @thousandhu @czy006 Could you help me review this patch?

File: deploy/kubernetes/operator/pkg/.gitkeep [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
```
Comment (jerqi) [inline deploy/kubernetes/operator/pkg/.gitkeep:1]: If operator directory have files, could we remove this file?

Comment (wangao1236) [reply deploy/kubernetes/operator/pkg/.gitkeep:1]: Yes

File: deploy/kubernetes/operator/api/uniffle/v1alpha1/remoteshuffleservice_types.go [hunk 2]
```diff
@@ -18,49 +18,303 @@
...
+
+	// SpecificNames indicates the specific pod names of shuffle servers which we want to upgrade.
+	// +optional
+	SpecificNames []string `json:"specificNames,omitempty"`
+}
+
+// RSSAutoscaler describes the desired functionality of the HPA object to be generated,
+// which automatically manages the replica count of any resource implementing the scale
+// subresource based on the metrics specified.
+type RSSAutoscaler struct {
```
[the remark this answers is not in the document]
Comment (wangao1236) [reply deploy/kubernetes/operator/api/uniffle/v1alpha1/remoteshuffleservice_types.go]: Ok, after discussion, I will remove this field first and add it in the future according to the requirements.

**Comment by thousandhu** [comment]
LGTM , thanks @wangao1236

**Review by jerqi — APPROVED** [review]
LGTM, thanks @wangao1236 @thousandhu , @zuston @czy006 if you have another suggestion, you can raise a new pr.

**Pull request merged**